csharp/ql/test/library-tests/dataflow/tuples/MixedPatterns.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+public class C
+{
+    public void M()
+    {
+        Base a = new Item() { X = "taint source" };
+        var ret1 = a switch
+        {
+            Item("taint source", 1) { Y: 10 } z => Sink(z.X),       // [TRUE POSITIVE]
+            Item("not taint source", 1) { Y: 10 } z => Sink(z.X),   // [FALSE POSITIVE]
```

Comment:
I don't think this is a false positive, as far as data flow is concerned. This is instead something where I would expect a given data flow configuration to prune data flow edges using guards.

It is basically the same as
```csharp
var x = "tainted";
if (x == "not tainted")
    Sink(x);
```